src/backend/src/lib.rs

```diff
@@ -436,6 +439,22 @@ fn list_custom_tokens() -> Vec<CustomToken> {
     read_state(|s| s.custom_token.get(&stored_principal).unwrap_or_default().0)
 }
 
+async fn bitcoin_ecdsa_public_key_of(canister_id: Principal, network: BitcoinNetwork) -> Vec<u8> {
+    let name = read_config(|s| s.ecdsa_key_name.clone());
+    let derivation_path = network_to_derivation_path(network)?;
+    let (key, ) = ecdsa_public_key(EcdsaPublicKeyArgument {
+        canister_id: Some(canister_id),
```

Comment:
I don't think we should path the `canister_id` or what does it means if both caller and optional `canister_id` are equals? Are the keys derived differently?

Reply:
that part is only to adapt it to the arguments of `ecdsa_public_key`, because it needs to be Optional. But the function `bitcoin_ecdsa_public_key_of` per-se does not accept Optional.

Another way of doing it would be below, but I would prefer to avoid that:

```rust
let (key, ) = ecdsa_public_key(EcdsaPublicKeyArgument {
        canister_id: canister_id.unwrap_or(Principal::anonymous()),
        derivation_path,
[...]
```

Reply:
I think you misunderstand my comment. I think that `canister_id` should be set to `None`.

Unless there is a reason not to do so. If there is can you share the references?

Reply:
agreed! In the final PR it would have been that, but since I already pushed it here, you will see canister=None